module-4/backend/week_three.md

```diff
@@ -1,11 +1,60 @@
 ## Week Three - Module 4 Recap
 
-Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week. 
+Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week.
 
-Note: When you're done, submit a PR. 
+Note: When you're done, submit a PR.
 
-TBD  
+1. At a high level, what is Node?
+
+- Node is a Javascript that is capable of being written and run in a server environment.
+
+2. What is Express? What is Express similar to in the Ruby world?
+
+- Express is a framework that allows Node written applications to be used as web applications.
+
+3. How do we setup a route when creating an API with Node and Express? Please provide a code snippet.
+
+- The app (which is set to express along with other uses of functions) is set. The route is then set using the app. For example, a GET to a "Jobs" route would look like:
+
+```
+app.get('/jobs', callbackfunction() {})
+```
+
+4. What do we use Knex for?
+
+- Knex is used to connect server-side Node code to a database.
+
+5. How could you organize your code to follow the MVC design pattern in your Quantified Self project?
+
+- The Server.js file can be used as the controller/routes. Server.js would house the actual route definitions (GET, POST, etc.). A separate folder containing database queries for given resources can be used as the "Models". Views are handled by the front end AJAX and API calls. However, we could house those in a views subdirectory should we want to pull the two applications together.
```

Comment:
absolutely - you could even have controller files as well, each route in your `server.js` file could call a controller and an action which then calls your models 👍